src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -76,7 +68,140 @@ public function elementValidate($element, FormStateInterface $form_state, $form)
+      '#title' => $this->t('Other widgets'),
+      '#description' => $description,
+      '#prefix' => '<div id="og-group-ref-other-groups">',
+      '#suffix' => '</div>',
+      '#cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+      '#cardinality_multiple' => TRUE,
+      '#theme' => 'field_multiple_value_form',
+      '#field_name' => $this->fieldDefinition->getName(),
+      '#max_delta' => 1,
+    ];
+
+    $elements['add_more'] = [
+      '#type' => 'button',
+      '#value' => $this->t('Add another item'),
+      '#name' => 'add_another_group',
+      '#ajax' => [
+        'callback' => [$this, 'addMoreAjax'],
+        'wrapper' => 'og-group-ref-other-groups',
+        'effect' => 'fade',
+      ],
+    ];
+    $start_key = 0;
+    // todo: get the other groups.
+
+    if (!$form_state->get('other_group_delta')) {
+      $form_state->set('other_group_delta', $start_key);
+    }
+
+    // Get the trigger element and check if this the add another item button.
+    $trigger_element = $form_state->getTriggeringElement();
+
+    if ( $trigger_element['#name'] == 'add_another_group') {
+      // Increase the number of other groups.
+      $delta = $form_state->get('other_group_delta') + 1;
+      $form_state->set('other_group_delta', $delta);
+    }
+
+    // Add another auto complete field.
+    for ($i = 0; $i <= $form_state->get('other_group_delta'); $i++) {
+      $elements[$i] = $this->otherGroupsSingle($i);
+    }
+  }
 
-    // todo: Check the writing permission for the current user.
+  /**
+   * Generating other groups auto complete element.
+   *
+   * @param $delta
+   *   The delta of the new element. Need to be the last delta in order to be
+   *   added in the end of the list.
+   * @param EntityInterface|NULL $entity
+   *   The entity object.
+   * @return array
+   *   A single entity reference input.
+   */
+  public function otherGroupsSingle($delta, EntityInterface $entity = NULL) {
+    return [
+      'target_id' => [
+        '#type' => "entity_autocomplete",
+        '#target_type' => $this->fieldDefinition->getTargetEntityTypeId(),
+        // todo: fix the definition in th UI level.
+        '#selection_handler' => 'default:og',
+        '#selection_settings' => ['other_groups' => TRUE],
+        '#default_value' => $entity ? $entity : NULL,
+      ],
+      '_weight' => [
+        '#type' => 'weight',
+        '#title_display' => 'invisible',
+        '#delta' => $delta,
+        '#default_value' => $delta,
+      ],
+    ];
   }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function massageFormValues(array $values, array $form, FormStateInterface $form_state) {
+    $parent_values = $values;
+
+    // Get the groups from the other groups widget.
+    foreach ($form[$this->fieldDefinition->getName()]['other_groups'] as $key => $value) {
+      if (!is_int($key)) {
+        continue;
+      }
+
+      preg_match("/.+\(([\w.]+)\)/", $value['target_id']['#value'], $matches);
```

Comment:
Could add a named capture group, that always reads a bit nicer IMO, also you could use different delimiters to make it easier on the eye:

```
|.+\((?<id>[\w.]+)\)|
```